### ISSUE TITLE
SR-17-0000206 FD Elevator Guide sheet and Workflow Process revisit

### ISSUE THREAD

**Issue by RikiSjachrani-COM** [City-of-Madison/Development-Team-2#3]
FD Elevator Guide sheet and Workflow Process revisit - Fire would like to modify the workflow of the fire elevator records to be closer to routine inspections.

Reference 
\\Fps1\data1\Information Technology\Applications and Systems\Accela\Active Projects\Elevator Guide sheet and Workflow Process revisit

Tasks
- [x] Assign to Jane and Kim
- [ ] Scripting
- [ ] Configuration

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM if we could possibly meet earlier that would work, this kind of jumped up on the priorities list.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM  I'm leaving now for my appointment. I should be back by 2:30p, possibly earlier. We can meet as sooon as I return.

**Comment by KimGrittner-COM** [comment]
I am available anytime this afternoon as well.

**Comment by RikiSjachrani-COM** [comment]
Can you please install from \\sccm\packages\One Note 2013

**Comment by RikiSjachrani-COM** [comment]
@KimGrittner-COM and @JaneSchneider-COM @ do you want to meet more on this, is this progressing since we last met?

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: @KimGrittner-COM and I are meeting at 1:30pm to discuss things. I was planning to start work on the scripting today, as I did not have time last week. Feel free to join us for meeting if you are available and so inclined.

**Comment by KimGrittner-COM** [comment]
Jane and I will review configuration at 1:30 today - see change log below:
 Only question I have is that the FIREELEV has several existing Inspection types (such as "Installation"), do we leave those alone or delete? (I left them for now)

Change Log on Accela Dev:
1.Updated AA Admin->Admin Tools->Inspection->Checklist Group
 a. FIREELEV
 b. Created FIREELEVREINSP
2.Created AA Admin->Admin Tools->Inspection->Checklist
 a. Fire Elevator Violations
3.Added AA Admin->Admin Tools-> Inspection->Inspection Result Group
 a. Added FIREELEVANNUAL
 b. Added FIREELEV1STRINSP
 c. Added FIREELEV2DTRINSP
 d. Added FIREELEV3STRINSP
4.Added Inspection Types: AA Admin->Admin Tools-> Inspection->Inspection
 a. Annual Inspection
 b. First Reinspection
 c. Second Reinspection
 d. Final Reinspection

**Comment by RikiSjachrani-COM** [comment]
Kim, just leave them for now. I believe Jerry is going to be providing us with additional inspection flow process diagrams that will deal with this

**Comment by RikiSjachrani-COM** [comment]
You all still working on this? I'm going to try and update the specifications based on the workflow. Let me know if you have any questions

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Not sure what you mean? Please clarify.
@KimGrittner-COM has completed the configurations in dev and I added version 4.0 of the fee schedule. I just getting organized with the scripting.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Are you saying the current specs are not valid and that you will be changing them, so I should hold off any working on scripting?

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM Take a look at the one note specs, I'm just adding the additional inspection statuses he mentions, particularly the Extension Granted; Citation; etc.. most of these are Pending actions and my changes to specs will not invalidate anything existing.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM : OK, thanks for update.

**Comment by RikiSjachrani-COM** [comment]
@KimGrittner-COM Kim, below are the changes I made to the specifications, my changes are highlighted and I note them in the workflow diagram that we've been working off of. Please review the one note and make appropriate changes/additions

![image](https://user-images.githubusercontent.com/13784495/28848993-9af83228-76da-11e7-8667-2962056a960e.png)

**Comment by KimGrittner-COM** [comment]
I have made these mods on dev with new Inspection result group data. (Already had named the Inspection Type "First Reinspection")

**Comment by RikiSjachrani-COM** [comment]
Ok, I'm going to give Jerry a call sometime this afternoon, did you have any questions to bring up with him? There's a big Fire Meeting tomorrow so I want him to know that we are working on this

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM:  I have questions, but not sure if for Jerry or just for you. I'd like to meet with you today to discuss.

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM change in scope!!!! the reinspection fees are all full price!!!!!

**Comment by RikiSjachrani-COM** [comment]
Ok, so I met with Jerry and went over all our questions. A couple changes and a lot of our assumptions are correct. I'll summarize the specification sheets and note where changes are. I made notes in the one note but I need to clarify some things....

**Comment by KimGrittner-COM** [comment]
I have moved the configuration stuff from Dev to Test.

**Comment by RikiSjachrani-COM** [comment]
@KimGrittner-COM, I'm just going through the configuration changes and everything is looking great! One thing I noticed is the Checklist Item Comment is not visible. This is under each checklist item at the far right under edit, the Comment Visible should be enabled. This is not too apparent from the checklist spreadsheet, but I know when Jerry didn't have a default comment, he still meant to display the comment box.

When you have a moment, will you please enable the Comment Visible on all the checklist items?

![image](https://user-images.githubusercontent.com/13784495/28979100-145faf1c-790e-11e7-9db7-da5aeb5a230b.png)

**Comment by KimGrittner-COM** [comment]
I have finished checking all of the "Comment Visible" checklist items on both dev and test.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Would you please verify or correct a few things for me?

1. The new FP_PTOISSUED set will be scheduled to run daily, (Mon-Fri) like the  FP10DAY, FPWRITTEN, and FPVERBAL sets, correct?

2. In Accela Report Manager, should both new "Fire Invoice" report (single & set) and the new "Fire Order" report (single & set) be configured under "Fire" or "Fire Prevention"? I'm thinking both would be under Fire Prevention, but want to confirm.

3.  Is the inspection/reinspection invoice set called "FPREINSP"?  

4. If #3 is correct
   a. Will this set continue to run weekly at same time (Fridays at 5:00am)? 

   b.  I was thinking that it would be helpful/more consistent to change the name of the set from  FPREINSP to something like FPINVOICE and the batch job name from "FD Reinsp Invoice" to "FD Invoice" or "FD Routine and Elev Invoice". Would that be OK? If you have any naming preference for set and/or batch job, please let me know.

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM in response to your numbered questions

1. Correct
2. I would say place these newer reports under the Fire Prevention directory. Question about the new "Fire Order" report, 
A. Is this suppose to replace all three order report types? I didn't realize we were consolidating them into one report and they had special text for written, verbal and 10-day
3. Yes
4. Yes
A. Yes
B. That makes sense and I am OK with renaming them to be more clear

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Yes, according to @BrendenPautsch-COM, the new "Fire Order" report (both single and set) replaces 10-Day, Verbal, and Written (and ElevatorOfficialNotice). The single version of the report takes two parameters: AltID and LetterType, where LetterType is one of 'FPWRITTEN, 'FPVERBAL, or 'FP10DAY', the corresponding SetID. The set version just has the SetID param, which tells the report which version to use.

@BrendenPautsch-COM: If any of what I have written above is not correct, please provide correct info.
Also, please move the "Fire Invoice" reports(single & SET)  from /Fire to /Fire Prevention, both in SSRS and in Accela. Also, please config the "Fire Order" and "Fire Oder SET" reports in /Fire Prevention in Accela Report Mgr once the reports are ready. Thanks!

**Comment by RikiSjachrani-COM** [comment]
Ok, that makes sense and that cool we were able to combine the reports. Once these have been setup in Dev, I will run through some front end testing against the sets and individual records.

**Comment by BrendenPautsch-COM** [comment]
Jane,

Everything you stated is correct.  I have also just completed moving the Fire Invoice and Set to the Fire Prevention folder and added Fire Orders and Set to Fire prevention as well.

I set up the individual reports to use the current AltID and I also created a variable for the Orders so you can select the letter type.  Let me know if youhave any issues.

Brenden

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM I'm looking the script table in the elevator annual inspection workflow with detail page in one note and am wondering if the box on the right should be what I'm testing against? 

My plan is to test all those scenarios and ensure the Action comes out correct. Is that accurate as far as what you scripted to?

Thanks,

Riki

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM what event will place a elevator record into the FPINVOICE Set? Also, I'm wondering if the PTO Expiration Date just be incremented after a Successful PTO Fee Assessed occurs? There's no rush on this, I'm just putting my thoughts somewhere

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Via RBIZDOMAIN search, I found these scripts that referenced the FPREINSP set, which I replaced with FPINVOICE:
- CMN:Permitting/Fire/*/*:InspectionStatused
- IRSA:Enforcement/Fire/Prevention/*
- CMN:Permitting/Fire/*/*:Schedule Reinspection

If I'm understanding your question about PTO Expiration Date: see CMN:Permitting/Fire/Elevator/*:CalcNextAnnualInspec, which is called from CMN:Permitting/Fire/Elevator/*:InspecApproved. 
If I'm not answering your question, please clarify.

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM 

About the FPINVOICE - I think this works for Prevention records but I don't see a way that Elevator records get dropped in there. I think it is just an oversight when I was spec'ing it

About the PTO Expiration Date - I see where it calculates ahead the 10 months for the next annual inspection but it doesn't increment the actual ASI Office Use Only field PTO Date to the next year, so the PTO Report is still the same date...Let's discuss this because I think it is just a oversight when I was spec'ing this

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Ah, OK. Yep, I missed these two items. 
I'll work on the PTO Expire Date first, as I understand that one, I think. 
We should briefly meet to make sure I understand when Elevator gets added to the FPINVOICE set.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: I think I have the ASI PTO Exp Date updating correctly now, but have question. 
I scheduled short meeting today at 4pm to discuss this and when to add to FPINVOICE set.
Also, discovered something about fees on Fire/Elevator record creation that I'd like to discuss.

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: OK, per our meeting, Fire/Elevator will be added to FPINVOICE in these two scripts:
- CMN:Permitting/Fire/Elevator/*:InspecApproved
- CMN:Permitting/Fire/Elevator/*:InspecDenied
when fees are assessed & invoiced.

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM I statused a second inspection on ELEV-2008-513818 and it added the re-inspection invoice, added it to the 10-Notice Set and schedule the Final Inspection but did not add to the FPINVOICE as I was expecting. When you have a moment, can to check that this is occuring or let me know if it shouldn't be occuring

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM : What environment are you in? Did you delete the inspection, because I'm not seeing it in Dev? Are you saying the record went into the FPREINSP set instead of the FPINVOICE set?

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: OK, I just tried this for ELEV-2008-924327 in Dev, and the record was added to the FPINVOICE set. 
I also just statused the Second Reinspection for ELEV-2008-513818 in Dev to Not Corrected, and the record was added to the FPINVOICE set, along with the being added to the FP10Day set, the reinspection fee being added, and the Final Reinspection being scheduled.
I didn't change anything, so not sure what happened when you tried it.

**Comment by RikiSjachrani-COM** [comment]
@JaneSchneider-COM yeah, it's in there and most likely was and I just did not refresh the portlet or something. sorry for the false alarm

**Comment by JaneSchneider-COM** [comment]
@RikiSjachrani-COM: Actually, the one entry in FPINVOICE for ELEV-2008-513818 is mine. I checked the SetDetails table, and the REC_DATE & REC_NAM are 2017-10-10 16:26:51.947 & ITJBS. 
So, still not sure what happened on your transaction. But, as I said above, when I statused an Annual Inspection and First or Second Reinspection with a status of Written Orders or Not Corrected, the record goes into the FPINVOICE set. 
Maybe try what you attempted again, and if you want me to watch when you do it, let me know.
There may be something not working, I'm just not clear on the sequence of events.

**Comment by RikiSjachrani-COM** [comment]
So we got a verbal sign off on this and can move everything to production, I'm going to send a confirmation email to Bill and Jerry confirming this. When do you think people would have time to promote everything to production?

**Comment by JaneSchneider-COM** [comment]
Am I correct that we cannot do tomorrow, as is Friday? So, that would leave today or next week, correct?
If this is a high priority, then I'll make time today.

**Comment by RikiSjachrani-COM** [comment]
We cannot do this Friday and we have a bunch of stuff to move. I've added Brenden since he has the report side. Maybe we can shoot for next week, just call a day...We should also submit a change request, I can do that today

**Comment by JaneSchneider-COM** [comment]
Kim & I were thinking Tuesday. 
How about I create a meeting for 9:00am where we can come up with a game plan of who will do what, and then we go off and do our assigned tasks. 
I was thinking I would move all the scripts.
Kim was thinking he would do all tasks related to guidesheets.
Brenden would set up the reports.
And then I could help Brenden set up the batch jobs and sets, if you would like.
Thoughts?

**Comment by RikiSjachrani-COM** [comment]
@BrendenPautsch-COM I am wondering if we could alter the calculation of the new invoice where it has "Previously Invoiced" since these are now going out weekly instead of daily. See the most recent example below, could we change it to a week? Would that make more sense?

![image](https://user-images.githubusercontent.com/13784495/31823346-8ece255e-b571-11e7-8aa1-cb470ab41245.png)

**Comment by BrendenPautsch-COM** [comment]
Riki,

I updated the query from
~~~~
WHEN dateadd(D,1,X.GF_FEE_APPLY_DATE) <= getdate() THEN 'Previously Invoiced'

TO

WHEN dateadd(D,7,X.GF_FEE_APPLY_DATE) <= getdate() THEN 'Previously Invoiced'

~~~~

That should fix it.  I ran the report individually and the text appears correct (No Text)

Brenden
